nilearn/regions/region_extractor.py

```diff
@@ -67,6 +68,61 @@ def _threshold_maps_ratio(maps_img, threshold):
     return threshold_maps_img
 
 
+def _remove_small_regions(input_data, mask_data, index,
+                          affine, min_size):
+    """Remove small regions in volume from input_data of specified min_size.
+
+    min_size should be specified in mm^3 (region size in volume).
+
+    Parameters
+    ----------
+    input_data : numpy.ndarray
+        Values inside the regions defined by labels contained in input_data
+        are summed together to get the size and compare with given min_size.
+        For example, see scipy.ndimage.label
+
+    mask_data : numpy.ndarray
+        Data should contains labels assigned to the regions contained in given
+        input_data to tag values to work on. The data must be of same shape of
+        input_data.
+
+    index : numpy.ndarray
+        A sequence of label numbers of the regions to be measured corresponding
+        to input_data. For example, sequence can be generated using
+        np.arange(n_labels + 1)
+
+    affine : numpy.ndarray
+        Affine of input_data is used to convert size in voxels to size in
+        volume of region in mm^3.
+
+    min_size : float in mm^3
+        Size of regions in input_data which falls below the specified min_size
+        of volume in mm^3 will be discarded.
+
+    Returns
+    -------
+    out : numpy.ndarray
+        Data returned will have regions removed specified by min_size
+        Otherwise, if criterion is not met then same input data will be
+        returned.
+    """
+
+    _, region_sizes = np.unique(input_data, return_counts=True)
+    size_in_mm_3 = min_size * np.abs(np.linalg.det(affine[:3, :3]))
```

Comment:
Oh, no, you want to compare this value to voxels counts, so what you meant is probably:
`size_in_vox = min_size / np.abs(np.linalg.det(affine[:3, :3]))`